core/integration/kafka/plugin.go

```diff
@@ -0,0 +1,271 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2022 CERN and copyright holders of ALICE O².
+ * Author: Piotr Konopka <piotr.jan.konopka@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+//go:generate protoc --go_out=. protos/kafka.proto
+
+package kafka
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/AliceO2Group/Control/common/utils/uid"
+	"github.com/AliceO2Group/Control/core/integration"
+	kafkapb "github.com/AliceO2Group/Control/core/integration/kafka/protos"
+	"github.com/AliceO2Group/Control/core/workflow/callable"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/golang/protobuf/proto"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+type Plugin struct {
+	kafkaBroker   string
+	kafkaProducer *kafka.Producer
+	envsInRunning map[string]*kafkapb.EnvInfo // env id is the key
+}
+
+func NewPlugin(endpoint string) integration.Plugin {
+	_, err := url.Parse(endpoint)
+	if err != nil {
+		log.WithField("endpoint", endpoint).
+			WithError(err).
+			Error("bad service endpoint")
+		return nil
+	}
+
+	return &Plugin{
+		kafkaBroker:   endpoint,
+		kafkaProducer: nil,
+	}
+}
+
+func (p *Plugin) GetName() string {
+	return "kafka"
+}
+
+func (p *Plugin) GetPrettyName() string {
+	return "Kafka FSM transition information service"
+}
+
+func (p *Plugin) GetEndpoint() string {
+	return viper.GetString("kafkaBroker")
+}
+
+func (p *Plugin) GetConnectionState() string {
+	if p == nil || p.kafkaProducer == nil {
+		return "UNKNOWN"
+	} else {
+		return "READY"
+	}
+}
+
+func (p *Plugin) GetData(environmentIds []uid.ID) string {
+	return ""
+}
+
+func (p *Plugin) FSMTransitionTopic(state string) string {
+	return "FSM_" + state
+}
+
+func (p *Plugin) ActiveRunsListTopic() string {
+	return "ACTIVE_RUNS"
+}
```

Comment:
@awegrzyn here are the topic names, let me know if you prefer something different.

Reply:
For me it's fine, if you want to go for "industry" standard you can take a look at: https://cnr.sh/essays/how-paint-bike-shed-kafka-topic-naming-conventions

Reply:
Actually it'd be the best if topics were configurable instead of hardcoded. On the other side there's advantage, if new state is introduced there's no need to change the code. If we keep it as it is, how do I find out full list of topics?

Reply:
As far as I understand, topic names are not something you should change often, because one cannot rename an existing one.
I am not sure if this is what you are looking for, but this lists all the topics:
```
[root@piotr-kafka ~]# /opt/kafka/kafka_2.12-2.8.1/bin/kafka-topics.sh --bootstrap-server localhost:9092 --list
ACTIVE_RUNS
FSM_CONFIGURED
FSM_DEPLOYED
FSM_RUNNIN
FSM_RUNNING
__consumer_offsets
notification
```
If we follow a more "industrial" convention, they should be quite easy to filter out.

Reply:
Okay indeed, this should be good enough

Reply:
Regarding the topic names, perhaps something like this instead?
- fsm transitions: `streaming.env_state.<state>` (notice that `state` is stored in upper case by aliecs, but i can convert it to lower case to adhere to the rest)
- active runs: `streaming.env_list.running`

Reply:
@awegrzyn :arrow_up:

Reply:
No worries, it can be upper case, my only comment is that `streaming` does bring any useful information, so I'd go for `aliecs.env.<state>` and `aliecs.env.running` or even skipping `aliecs.`.

Reply:
Agreed about `streaming`, but I believe we should differentiate between `env_state` and `env_list`. The first provides updates about new states, the latter contains lists of environments in given state (now only `running`).
If we used `env`, we would have `aliecs.env.RUNNING` for new running state notification and `aliecs.env.running` for the list of environments in `RUNNING`

Reply:
ok, indeed